src/types/api.js

```diff
@@ -5,6 +5,22 @@
  * https://h.readthedocs.io/en/latest/api-reference/
  */
 
+/**
+ * An entry in the API index response (`/api`) describing an API route.
+ *
+ * @typedef RouteMetadata
+ * @prop {string} method - HTTP method
+ * @prop {string} url - URL template
+ * @prop {string} desc - Description of API route
+ */
+
+/**
+ * Structure of the `links` field of the API index response (`/api`) describing
+ * available API routes.
+ *
+ * @typedef {{ [key: string]: RouteMap|RouteMetadata }} RouteMap
```

Comment:
I tried using `Record<string, RouteMap|RouteMetadata>` here but that turns out not to support the recursive nature of this type.

Reply:
Understand